Sharpasonne/Engine.cs

```diff
@@ -15,17 +15,28 @@ public class Engine : IEngine
     {
         public Board Board { get; } = new Board();
 
-        public IImmutableDictionary<Type, IImmutableList<IRule<IGameAction>>> Rules { get; }
-            = ImmutableDictionary<Type, IImmutableList<IRule<IGameAction>>>.Empty;
+        /// <summary>
+        /// Players collection for managing player stats and turns.
+        /// </summary>
+        public Players Players { get; }
 
         /// <summary>
-        /// 
+        /// 1-index number of the player who's turn it is to play an action.
         /// </summary>
+        public int CurrentPlayerTurn { get; } = 1;
```

Comment:
Have you considered a queue/stack of players that rotates at each turn? Or maybe an object that kinda does that for you. Keeping an index smells.

Reply:
I did not, I used an int to keep compatible with #13. We should raise a new issue for how to represent players after this.